### DISCUSSION TITLE
ilogtail-1.8.4关于收集K8S共享pvc内日志目录，tag错乱的问题

### DISCUSSION THREAD

**Discussion by ywops** [alibaba/ilogtail#1467] [Help] [answered]
1. 版本：ilogtail-1.8.4
2. 部署方式：DaemonSet
3. 问题场景：在node节点上的从本地磁盘创建pv/pvc并以hostpath 方式挂载到了容器内/data，用于收集容器应用打印日志。因此节点所有容器是共享/data的。但此时使用ilogtail采集出现了tag 和 具体日志目录不匹配的问题，
4. 猜测原因：由于/data共享，任何容器内都能看到所有其他容器的日志，导致tag 与实际日志内容错乱。

例如，elk中tag如下：
_tags.k8s.pod.name     hcapp-mcnrr
_tag.logs.file.path    /data/logs/pgapp/detail.log

正常情况下应该收集/data/logs/hcapp/detail.log

[ilogtail配置.txt](https://github.com/alibaba/ilogtail/files/15145531/ilogtail.txt)

**Comment by messixukejia** [comment]
采集配置是只有一个用于采集所有的应用？ 
如果不同应用的子目录能区分开，可以讲采集配置分开，每个应用有自己的采集配置独立采集各自的子目录。

**Comment by ywops** [comment]
LogPath定义为 /data/logs，此目录内的应用是区分子目录的，但是无法通过FilePattern  正则的方式通过一个采集配置文件区分下。 由于微服务应用数量过百，咱们FilePattern  是否支持变量的方式，与采集到的环境变量来组合匹配。

**Reply by messixukejia** [reply, marked as answer]
![image](https://github.com/alibaba/ilogtail/assets/13250519/1cb30742-b615-4555-8283-7a23f010f961)
enable_env_ref_in_config 开关是在开源版是默认开启的，所以上图的功能直接可用。

如果解决了问题，欢迎帮忙在如下文档提交个pr完善下，多谢。
https://github.com/alibaba/ilogtail/blob/main/docs/cn/configuration/collection-config.md

另外，也麻烦登记下使用场景。https://github.com/alibaba/ilogtail/issues/693
